src/phoenix/server/api/dataloaders/cache/two_tier_cache.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC
+from asyncio import Future
+from typing import Any, Callable, Generic, Optional, Tuple, TypeVar, cast
+
+from cachetools import Cache, LFUCache
+from strawberry.dataloader import AbstractCache
+
+_Key = TypeVar("_Key")
+_Result = TypeVar("_Result")
+
+_Section = TypeVar("_Section")
+_SubKey = TypeVar("_SubKey")
+
+
+class TwoTierCache(
+    AbstractCache[_Key, _Result],
+    Generic[_Key, _Result, _Section, _SubKey],
+    ABC,
+):
+    _main_cache_factory: "staticmethod[[], Cache[_Section, Cache[_SubKey, Future[_Result]]]]" = (
+        staticmethod(lambda: LFUCache(maxsize=32))
+    )
+    _sub_cache_factory: "staticmethod[[], Cache[_SubKey, Future[_Result]]]" = staticmethod(
+        lambda: LFUCache(maxsize=8)
+    )
+
+    def __init_subclass__(
+        cls,
+        *,
+        main_cache_factory: Optional[
+            Callable[[], "Cache[_Section, Cache[_SubKey, Future[_Result]]]"]
+        ] = None,
+        sub_cache_factory: Optional[Callable[[], "Cache[_SubKey, Future[_Result]]"]] = None,
+        **kwargs: Any,
+    ) -> None:
+        super().__init_subclass__(**kwargs)
+        if main_cache_factory:
+            cls._main_cache_factory = staticmethod(main_cache_factory)
+        if sub_cache_factory:
+            cls._sub_cache_factory = staticmethod(sub_cache_factory)
```

Comment:
I think this is overly abstract for what we need right now. It looks like you're anticipating that we will need alternative caching strategies, but all of the subclasses are currently using LFU caches with different maxsize parameters. It would be simpler to just have an `__init__` that takes a `maxsize`, wouldn't it?